src/plugins/vis_type_timeseries/public/application/components/aggs/top_hit.js

```diff
@@ -110,7 +110,7 @@ const TopHitAggUi = (props) => {
     PANEL_TYPES.METRIC,
     PANEL_TYPES.MARKDOWN,
   ].includes(panel.type)
-    ? [KBN_FIELD_TYPES.NUMBER, KBN_FIELD_TYPES.STRING]
+    ? [KBN_FIELD_TYPES.NUMBER, KBN_FIELD_TYPES.STRING, KBN_FIELD_TYPES.DATE]
     : [KBN_FIELD_TYPES.NUMBER];
 
```

Comment:
@DziyanaDzeraviankina @stratoula I see after switching from `numbers` to `strings` (and now to `dates`) our model keep wrong value (value which was selected to `numbers`). On UI we see `Select...` . This is issue should be fixed. 

![image](https://user-images.githubusercontent.com/20072247/124484222-d9ecee80-ddb3-11eb-817b-f06ffbc8421a.png)


Please think about adding UI effect to set first option from available list 

```
  useEffect(() => {
    const defaultFn = aggWithOptions?.[0].value;
    const aggWith = model[AGG_WITH_KEY];
    if (aggWith && defaultFn && aggWith !== defaultFn && !selectedAggWithOption) {
      handleChange({
        AGG_WITH_KEY: defaultFn,
      });
    }
  }, [model, selectedAggWithOption, aggWithOptions, handleChange]);
```

or fix it using different way, e.g. set `agg_with` to `noop`. I don't have strong opinion here